lib/ws-channel.js

```diff
@@ -56,43 +59,68 @@ WebsocketChannel.prototype._onNotification = function(message) {
 function nullCallback() {
 }
 
-WebsocketChannel.prototype.connect = function(uri, _token) {
-  var token = _token || url.parse(uri).auth;
+WebsocketChannel.prototype.connect = function(uri, options) {
+  assert(!this._clientToken);
+  assert(!this._sessionToken);
+  
+  options = options || {};
+  this._clientToken = options.clientToken || url.parse(uri).auth;
+  this._sessionToken = options.sessionToken || this._generateToken();
+  
+  this._uri = uri;
+  
+  return this._connect();
+}
 
-  this._debug = debug('ws-channel:' + token.substr(0, 5));
-  this._debug('connect: uri %j token %s)', uri, _token || '(from uri)');
 
-  assert(uri);
-  assert(token, 'token required');
+WebsocketChannel.prototype.reconnect = function() { 
+  return this._connect();
+}
 
-  this._uri = uri;
-  this._token = token;
+  
+WebsocketChannel.prototype._connect = function() {
+  var clientToken = this._clientToken;
+  var sessionToken = this._sessionToken;
+  var isNew = this._seq === -1 && this._receivedSeq === -1;
+  var uri = this._uri;
+  
+  this._debug = debug('ws-channel:' + clientToken.substr(0, 5));
```

Comment:
this statement and following need to go back in .connect(), that's where this._clientToken is assigned, and where its known if the token was passed as an arg, or parsed from the uri